solidity/contracts/converter/ConverterUpgrader.sol

```diff
@@ -240,23 +246,16 @@ contract ConverterUpgrader is IConverterUpgrader, ContractRegistryClient {
     function transferReserveBalancesVersion45(ILegacyConverterVersion45 _oldConverter, IConverter _newConverter)
         private
     {
-        uint256 reserveBalance;
         uint16 reserveTokenCount = _oldConverter.connectorTokenCount();
-
         for (uint16 i = 0; i < reserveTokenCount; i++) {
-            IERC20 reserveAddress = _oldConverter.connectorTokens(i);
-            // Ether reserve
-            if (reserveAddress == NATIVE_TOKEN_ADDRESS) {
-                if (address(_oldConverter).balance > 0) {
+            IReserveToken reserveToken = _oldConverter.connectorTokens(i);
+
+            uint256 reserveBalance = reserveToken.balanceOf(address(_oldConverter));
+            if (reserveBalance > 0) {
+                if (reserveToken.isNativeToken()) {

```

Comment:
technically, there's no old ETH converter in the registry anymore, so we can just use the new interface here.
if someone tries to upgrade an old converter, it'll fail. but it's still possible to withdraw liquidity out.

Reply:
I'll implement it in a separate PR